sst/include/SimengCoreWrapper.hh

```diff
@@ -0,0 +1,187 @@
+ * SST simulation. The SimEng core as well as componets/interfaces from SST
+ * required to ensure a succesful integration are instantiated and configured in
+ * this class as well. This class acts as the point of main contact for clock
+ * ticks recieved from SST and hence is also responsible for ticking the SimEng
+ * core and other classes assosciated to it.
+ */
+class SimengCoreWrapper : public SST::Component {
+ public:
+  SimengCoreWrapper(SST::ComponentId_t id, SST::Params& params);
+  ~SimengCoreWrapper();
+
+  /** SST lifecycle methods (in-order of invocation) overriden from
+   * SST::Component. */
+
+  /**
+   * This is the init lifecycle method present in all SST::Components.
+   * Here it is overriden to include init calls to all other SST::Components
+   * which are contained inside SimengCoreWrapper. It is neccessary call all
+   * lifecycle methods for SST::Component(s).
+   */
+  void init(unsigned int phase);
+
+  /**
+   * This is the setup lifecycle method present in all SST::Components.
+   * Here it is overriden to include setup calls to all other SST::Components
+   * which are contained inside SimengCoreWrapper. It is neccessary call all
+   * lifecycle methods for SST::Component(s).
+   */
+  void setup();
+
+  /**
+   * This is the finish lifecycle method present in all SST::Components.
+   * Here it is overriden to finish statistics about the SimEng simulation.
+   */
+  void finish();
+
+  /**
+   * The clockTick is a method present in all SST::Components. This fuction
+   * is called everytime the SST clock ticks. The current clock cycle is passed
+   * as an argument by SST. The SimEng core ticks in this method.
+   */
+  bool clockTick(SST::Cycle_t currentCycle);
+
+  /**
+   * This handle event method is registered to StandardMem interface is called
+   * everytime a memory request is forwarded by the interface. This functions
+   * acts a callback and invokes SimengMemHandlers on the memory request.
+   */
+  void handleEvent(StandardMem::Request* ev);
+
+  /**
+   * SST supplied MACRO used to register custom SST:Components with
+   * the SST Core.
+   */
+  SST_ELI_REGISTER_COMPONENT(SimengCoreWrapper, "sstsimeng", "simengcore",
+                             SST_ELI_ELEMENT_VERSION(1, 0, 0),
+                             "Simeng core wrapper for SST",
+                             COMPONENT_CATEGORY_PROCESSOR)
+
+  /**
+   * SST supplied MACRO used to document all parameters needed by
+   * a custom SST:Component.
+   */
+  SST_ELI_DOCUMENT_PARAMS(
+      {"config_path", "Path to Simeng YAML config file (string)", ""},
+      {"executable_path",
+       "Path to executable binary to be run by SimEng (string)", ""},
+      {"executable_args",
+       "argument to be passed to the executable binary (string)", ""},
+      {"clock", "Clock rate of the SST clock (string)", ""},
+      {"max_addr_memory", "Maximum address that memory can access (int)"}, )
+
```

Comment:
I have forgotten to document `cache_line_width`  via ELI. Can you please add an entry for it? Thanks!